ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -1123,11 +1127,29 @@ private void verifyNumQueries(final int numQueries) {
   }
 
   private void givenConnectorExists() {
+    // Make sure we are starting from a clean slate before creating a new connector.
+    cleanupConnectors();
+
     makeKsqlRequest("CREATE SOURCE CONNECTOR " + TEST_CONNECTOR + " WITH ('connector.class'='" + MOCK_SOURCE_CLASS + "');");
+
+    assertThatEventually(
+        () -> {
+          try {
+            return ((ConnectorList) makeKsqlRequest("SHOW CONNECTORS;").get(0)).getConnectors().size();
+          } catch (AssertionError e) {
```

Comment:
Same questions as above. I notice there's no analogous try-catch around the `SHOW CONNECTORS;` request in the `cleanupConnectors()` helper method, curious why this situation is different.

Reply:
It takes a while to provision a connector, even after `CREATE CONNECTOR` returns a successful response. When ksqlDB calls `SHOW CONNECTORS;`, it makes a call to '/info' for reach connector, so if the new connector is not done provisioning, it returns an error response which propagates.